### DISCUSSION TITLE
lib.detect库可以用于声明域吗

### DISCUSSION THREAD

**Discussion by jingkaimori** [xmake-io/xmake#4243] [Q&A] [answered]
先检测头文件是否存在，然后根据配置选项和检测结果来决定变量的值，声明域的`check_cincludes`不返回检测结果，也不能通过字符串拼接从内部选项读这个结果。
```lua
local has_stdlib = check_cincludes("stdlib.h")
if has_stdlib and option("usexxx") then
     set_configvar("USEXXX", 1)
end
```

**Comment by waruqi** [comment, marked as answer]
描述域仅用于配置，不要做这种脚本操作，不支持

描述域的 check_cincludes 也就只是个配置 option 的辅助接口，这个阶段 option 都还没配置全，更别说获取option 结果了
